File: api/contrib/consultar/consultar.php
<?php
/*
 * Copyright (C) 2017-2025 CRLibre <https://crlibre.org>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published
 * by the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

function consutar()
{
    $curl   = curl_init();
    $clave  = params_get('clave');

    if ($clave == "" || strlen($clave) == 0)
        return "La clave no puede ser en blanco";

    $url = null;
    if (params_get("client_id") == 'api-stag')
        $url = "https://api-sandbox.comprobanteselectronicos.go.cr/recepcion/v1/recepcion/";
    else if (params_get("client_id") == 'api-prod')
        $url = "https://api.comprobanteselectronicos.go.cr/recepcion/v1/recepcion/";

    if ($url == null)
        return "Ha ocurrido un error en el client_id.";

    curl_setopt_array($curl, array(
        CURLOPT_URL             => $url . $clave,
        CURLOPT_RETURNTRANSFER  => true,
        CURLOPT_ENCODING        => "",
        CURLOPT_MAXREDIRS       => 10,
        CURLOPT_SSL_VERIFYHOST  => 0,
        CURLOPT_SSL_VERIFYPEER  => 0,
        CURLOPT_TIMEOUT         => 30,
        CURLOPT_HTTP_VERSION    => CURL_HTTP_VERSION_1_1,
        CURLOPT_CUSTOMREQUEST   => "GET",
        CURLOPT_HTTPHEADER      => array(
            "Authorization: Bearer " . params_get('token'),
            "Cache-Control: no-cache",
            "Content-Type: application/x-www-form-urlencoded",
            "Postman-Token: bf8dc171-5bb7-fa54-7416-56c5cda9bf5c"
        ),
    ));

    $response   = curl_exec($curl);
    $status     = curl_getinfo($curl, CURLINFO_HTTP_CODE);
    $err        = curl_error($curl);
    curl_close($curl);

    if ($err)
    {
        $arrayResp = array(
            "Status" => $status,
            "to" => $apiTo,
            "text" => $err
        );
        return $arrayResp;
    }
    else
    {
         $response = json_decode($response);
        return $response;
    }
}

?>

File: python-api/python_api/handlers/consultar.py
from __future__ import annotations

import httpx
from fastapi import Request
from fastapi.responses import JSONResponse

from python_api.responses import tools_reply_compatible


def _consultar_url(client_id: str) -> str | None:
    if client_id == "api-stag":
        return "https://api-sandbox.comprobanteselectronicos.go.cr/recepcion/v1/recepcion/"
    if client_id == "api-prod":
        return "https://api.comprobanteselectronicos.go.cr/recepcion/v1/recepcion/"
    return None


async def consultarCom(_: Request, params: dict[str, str]) -> JSONResponse:
    clave = str(params.get("clave", ""))
    token = str(params.get("token", ""))
    client_id = str(params.get("client_id", ""))

    if clave == "":
        return tools_reply_compatible("La clave no puede ser en blanco")

    url_base = _consultar_url(client_id)
    if url_base is None:
        return tools_reply_compatible("Ha ocurrido un error en el client_id.")

    headers = {
        "Authorization": f"Bearer {token}",
        "Cache-Control": "no-cache",
        "Content-Type": "application/x-www-form-urlencoded",
        "Postman-Token": "bf8dc171-5bb7-fa54-7416-56c5cda9bf5c",
    }

    try:
        async with httpx.AsyncClient(timeout=30, verify=False) as client:
            response = await client.get(f"{url_base}{clave}", headers=headers)
    except Exception as exc:
        return tools_reply_compatible({"Status": 0, "to": None, "text": str(exc)})

    try:
        data = response.json()
    except ValueError:
        data = None
    return tools_reply_compatible(data)

